lib/storage.js

```diff
@@ -39,17 +40,23 @@ function defaultStorage (storage, opts) {
   try {
     // Store in .dat with secret in ~/.dat
     if (fs.statSync(storage).isDirectory()) {
-      return datStore(storage)
+      if (typeof opts.secretDir !== 'undefined') {
+        fs.statSync(opts.secretDir).isDirectory()
+      }
+      return datStore(storage, opts)
     }
   } catch (e) {
     // Does not exist, make dir
     try {
       fs.mkdirSync(storage)
```

Comment:
Can you either separate these two try/catch out or check which directory errored. For example, if `fs.statSync(opts.secretDir).isDirectory()` errors, then we'll make the storage directory even if it exists.

Reply:
Yep reworked.